sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -553,13 +558,43 @@ private <T> Mono<CosmosItemResponse<T>> createItemInternal(T item, CosmosItemReq
                 getId(),
                 database.getId(),
                 database.getClient(),
-                ModelBridgeInternal.getConsistencyLevel(effectiveOptions),
+                effectiveOptions.getConsistencyLevel(),
                 OperationType.Create,
                 ResourceType.Document,
                 requestOptions,
                 trackingId);
     }
 
+    private void applyPolicies(OperationType operationType, ResourceType resourceType, OverridableRequestOptions requestOptions,
+                               String spanName) {
+        CosmosAsyncClient client = this.database.getClient();
+        CosmosDiagnosticsThresholds thresholds = requestOptions != null
+            ? clientAccessor.getEffectiveDiagnosticsThresholds(client, requestOptions.getDiagnosticsThresholds())
+            : clientAccessor.getEffectiveDiagnosticsThresholds(client, null);
+        CosmosDiagnosticsContext cosmosCtx = ctxAccessor.create(
+            spanName,
+            clientAccessor.getAccountTagValue(client),
+            BridgeInternal.getServiceEndpoint(client),
+            database.getId(),
+            getId(),
+            resourceType,
+            operationType,
+            null,
+            clientAccessor.getEffectiveConsistencyLevel(client, operationType, requestOptions.getConsistencyLevel()),
+            null,
+            thresholds,
+            null,
+            clientAccessor.getConnectionMode(client),
+            clientAccessor.getUserAgent(client),
+            null,
+            null,
+            requestOptions);
```

Comment:
should we also allow queryName to be part of the CosmosDiagnosticsContext, vaguely remember there might be a use case to have different configuration based on different custom query name

Reply:
yes, good point. It is the most common denominator to distinguish between queries that are to be expected to be processed quickly or slowly - because it is kind of a logical identified for a specific type of query

Reply:
o, could be handy in the global override callback